src/nvim/edit.c

```diff
@@ -4239,9 +4239,10 @@ void ins_compl_check_keys(int frequency)
 static int ins_compl_key2dir(int c)
 {
   if (c == Ctrl_P || c == Ctrl_L
-      || (pum_visible() && (c == K_PAGEUP || c == K_KPAGEUP
-                            || c == K_S_UP || c == K_UP)))
+      || c == K_PAGEUP || c == K_KPAGEUP
+      || c == K_S_UP || c == K_UP) {
```

Comment:
Ok, it solves that particular case. But this changes the value of `compl_direction`, which is used in `ins_compl_dictionaries` and `ins_compl_get_exp` (in about 10 different places... what a mess).

If we start getting reports of the wrong item being selected, we will need to revisit this...